### ISSUE TITLE
Confusion about loss calculation with loss_factory api

### ISSUE THREAD

**Issue by Dongzhou-1996** [CVLAB-Unibo/Real-time-self-adaptive-deep-stereo#65]
I wanna use 'mean SSIM L1' function to evaluate the accuracy of disparity map predicted by MADNet and ZED camera. Those disparity map have been written down in .txt file. However, I found that the running time of loss function is extremely slow, about 0.25 fps. I think my code runs with GPU indeed. Looking forward to your reply!

kindly regards,
dong zhou

part of code is attached below:
`````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````
disparity_map = np.loadtxt(disparity_map_path, dtype=np.float, delimiter=',')[np.newaxis, :, :, np.newaxis]
print('=> disparity map shape: {}'.format(disparity_map.shape))
tf_disparity_map = [tf.constant(disparity_map, dtype=tf.float32)]
# mean SSIM L1 loss calculation
start_time = time.time()
loss = loss_factory.get_reprojection_loss('mean_SSIM_l1')(tf_disparity_map, inputs)
end_time = time.time()
fps = 1.0/(end_time - start_time)
print('=> the speed of mean SSIM L1 loss calculation: {:04f}'.format(fps))
loss = session.run(loss)
if disparity_type == 'zed':
    print('=> the disparity estimation loss of zed: {:0.6f}'.format(loss))
else:
    print('=> the {} disparity estimation loss with {}: {:0.6f}'.format(disparity_type, mode, loss))
``````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````

part of outputs:
`````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````
=> calculate disparity losses for sequence 0002
=> Successfully loaded 0th frame in sequence 0002
=> left roi image shape: (1, 204, 311, 3)
=> disparity map shape: (1, 204, 311, 1)
=> the speed of mean SSIM L1 loss calculation: 0.239906
=> the global disparity estimation loss with FULL: 0.050883

=> Successfully loaded 1th frame in sequence 0002
=> left roi image shape: (1, 203, 306, 3)
=> disparity map shape: (1, 203, 306, 1)
=> the speed of mean SSIM L1 loss calculation: 0.255643
=> the global disparity estimation loss with FULL: 0.062041

=> Successfully loaded 2th frame in sequence 0002
=> left roi image shape: (1, 203, 305, 3)
=> disparity map shape: (1, 203, 305, 1)
=> the speed of mean SSIM L1 loss calculation: 0.257102
=> the global disparity estimation loss with FULL: 0.050178

=> Successfully loaded 3th frame in sequence 0002
=> left roi image shape: (1, 204, 303, 3)
=> disparity map shape: (1, 204, 303, 1)
=> the speed of mean SSIM L1 loss calculation: 0.255092
=> the global disparity estimation loss with FULL: 0.045681

=> Successfully loaded 4th frame in sequence 0002
=> left roi image shape: (1, 201, 304, 3)
=> disparity map shape: (1, 201, 304, 1)
=> the speed of mean SSIM L1 loss calculation: 0.250545
=> the global disparity estimation loss with FULL: 0.041843
`````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````

**Comment by AlessioTonioni** [comment]
The time needed to compute the loss is the one needed to execute `session.run(loss)`, the time you are measuring is the time needed to build the tensor flow graph that will run the loss computation.
If you need to process multiple images, you will built the graph once and execute it multiple times with `sess.run`, so it might be that the graph creation needs some seconds.

**Comment by Dongzhou-1996** [comment]
> 
> 
> The time needed to compute the loss is the one needed to execute `session.run(loss)`, the time you are measuring is the time needed to build the tensor flow graph that will run the loss computation.
> If you need to process multiple images, you will built the graph once and execute it multiple times with `sess.run`, so it might be that the graph creation needs some seconds.

Thanks for your response! So sorry, I'am not familiar with TensorFlow. You mean that the api of loss_factory.get_reprojection_loss() only create the graph with ops, and I should run this graph with different inputs by using 'session.run(loss, feed_dict=inputs)'. 

Regards,
dong zhou

**Comment by AlessioTonioni** [comment]
Yes you can have the disparity map and the left right frames as `tf.placeholder` and then provide them to the loss function as you mentioned. 
Unfortunately this code was developed on tf 1.x so it's very likely that it will not work with eager execution style code.

**Comment by Dongzhou-1996** [comment]
Thanks again for your help, I had solved this issue as you just said! :D
